file_test.go

```diff
@@ -1050,3 +1050,103 @@ func TestFile__RemoveBatch(t *testing.T) {
 		t.Errorf("unexpected number of return entries: %d", len(file.ReturnEntries))
 	}
 }
+
+func TestFile__CalculateBalancedOffsetCredit(t *testing.T) {
+	// Create an ACH file with a debit and add a credit offset record
+	f := mockFilePPD()
+	if b, ok := f.Batches[0].(*BatchPPD); ok {
+		b.Header.ServiceClassCode = DebitsOnly
+		b.Entries[0].TransactionCode = CheckingDebit
+		if err := f.Batches[0].Create(); err != nil {
+			t.Fatal(err)
+		}
+	} else {
+		t.Fatalf("got %T: %#v", f.Batches[0], f.Batches[0])
+	}
+	if len(f.Batches) != 1 {
+		t.Errorf("found unexpected %d batches", len(f.Batches))
+	}
+	if err := f.Create(); err != nil {
+		t.Fatal(err)
+	}
+
+	bh := NewBatchHeader()
+	bh.StandardEntryClassCode = PPD
+	bh.CompanyName = "ACME Corporation"
+	bh.CompanyIdentification = "121042882"
+	bh.CompanyEntryDescription = "OFFSET"
+	bh.ODFIIdentification = "12104288"
+
+	ed := NewEntryDetail()
+	ed.SetRDFI("121042882")
+	ed.DFIAccountNumber = "123456789"
+	ed.IndividualName = "OFFSET"
+	ed.TransactionCode = CheckingCredit
```

Comment:
I don't like how we needed to specify Checking(Credit/Debit) as that defeats the purpose of this method (to automate determining that in the method), but I'm unsure how we could pick between Checking and Savings as we don't know the ODFI account types. Thoughts?

Reply:
Good question, I believe that was part discussion we had initially,  Let the caller do the work instead of creating a function do it.  So the caller creates a batch with the entry detail record that is the offset and adds the batch to file similar to any other batch being added to a file.

Reply:
Right, but I was hoping the caller didn't need to know about debit vs credit. It sorta defeats the purpose of this helper method if they have to account for it also.

Reply:
I agree